pkg/kubectl/resource/visitor.go

```diff
@@ -487,11 +487,12 @@ func (v *FileVisitor) Visit(fn VisitorFunc) error {
 		f = os.Stdin
 	} else {
 		var err error
-		if f, err = os.Open(v.Path); err != nil {
+		f, err = os.Open(v.Path)
+		defer f.Close()
```

Comment:
It does look safe: https://golang.org/src/os/file_unix.go?s=3603:3614#L119 but is this really the idiom? I see use use 'defer close before error check' once in kubectl and 3 times in client-go.
```
git grep -B 1 defer | grep -B1 Close | grep Open -A1 | grep -v vendor
```
I see a whole lot more 'defer after check for err' throughout the code base.